framework/integration/test/basic/basic_test.go

```diff
@@ -49,9 +49,27 @@ func Test_Finalizer_Integration_Basic(t *testing.T) {
 		t.Fatal("expected", nil, "got", err)
 	}
 
-	// We directly pass the configmap to UpdateFunc.
+	// We update the configmap with a meanless label to ensure a change in the
+	// ResourceVersion of the ConfigMap.
+	cm.SetLabels(
+		map[string]string{
+			"testlabel": "testlabel",
+		},
+	)
+	_, err = client.UpdateConfigMap(cm)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We directly pass the _old_ configmap to UpdateFunc.
+	// This is expected to fail, we want to make sure that we only use the latest
+	// ResourceVersion of an object.
```

Comment:
Ok, but how does the test verify that it fails? This is actually something we want to be assured of.

Reply:
This does not fail anymore, we just pass the old object here and verify that only the correct finalizers have been set.